Add the new global theme and some local files for new theme

budparr opened the pull request.

Commit:
swap in layouts from working dir

Commit:
add single templates

Commit:
mobile tweaks

Commit:
clean up unneeded static files

Commit:
update theme from docs site changes

Commit:
add global menu to config

Commit:
[wip] temp fix for limiting the number of tags on the mobile menu

Commit:
[wip] perf improvement

this might be questionable, and should be evaluated further.

Commit:
overflow didn't work

bep commented on layouts/index.html:
```diff
@@ -1,12 +1,5 @@
-{{ end }}
+{{ define "main" }}
+  {{ $section_to_display := first 1000 .Site.RegularPages.ByLastmod.Reverse }}
+  {{ $.Scratch.Add "section_to_display" $section_to_display }}
```
Even though this works, and I don't see how it should not continue to do so, concurrency-wise, but I consider this an anti-pattern. The Scratch is built for a particular flaw in the Golang templates, and this isn't it. Pass the values you need as arguments (use dict) to the partial.

budparr replied:
Good call, and just rushing on my part for the easiest way to solve the problem when I conceived of this pattern. I'll need to fix this in the gohugo theme, too.

bep commented on layouts/partials/pagelayout.html:
```diff
@@ -0,0 +1,33 @@
+        </div>
+      </article>
+      <div class="flex flex-wrap justify-left pr4">
+        {{ $globalVariables := .Site.Data.globals }}
```
Everything starting with "$" is by definition a variable, so no need to say it again, "$globals" is better.

budparr replied:
Good point, thanks.

bep commented on static/_headers:
```diff
@@ -1,4 +0,0 @@
-/*
```
Could you undelete this file?

budparr replied:
I had intended to move this into the theme so it could be used in all the sites. You prefer it local to the site?

bep replied:
Ah, OK -- I didn't "see the theme" when I wrote this. Let it be in the theme.

budparr replied:
And, just so you know, it's not in the theme yet. I didn't think to add it until I saw yours here.

bep replied:
OK.

bep commented on layouts/_default/single.html:
```diff
@@ -1,60 +1,52 @@
-            </div>
+{{ define "main" }}
+
+    {{ $currentPageUrl := .URL }}
```
Is this used? If so, the Permalink variants is probably what you want.

bep replied:
I will take make some notes about the theme tomorrow (https://github.com/budparr/gohugo-theme)

bep reviewed (changes requested):
In general it looks very good. I have added a coiple of coments, the main being the Scratch/Partial one. Nor-amally I would said fine, but people look to these projects for copy-and-paste inspiration, so it would be good to get it right before it spreads.

Commit:
Lose the scratch anti-pattern and make this more succinct

https://github.com/spf13/hugoThemesSite/pull/39#discussion_r118791728

Commit:
these aren't needed here

Commit:
These aren't needed here, unnecessary complexity

wrong message on the last commit

Commit:
remove scratch anti-pattern

Commit:
make the ishome text more succinct and give it context

Commit:
give the tag data context

Commit:
revert change. the white is for mobile only

Commit:
remove unused currentPageUrl and tighten up

https://github.com/spf13/hugoThemesSite/pull/39#discussion_r118792745

budparr commented:
I addressed the issues you mentioned (and thanks for the comments), and I'll add the relevant changes to the theme as well.

Commit:
update theme from work on hugodocs site

bep commented:
The "Tag:" logic seems to be slightly broken. I tried to push a fix to this branch, but no access.

budparr commented:
What's the issue/fix? I'm happy to do it so I can learn from doing so.

bep commented:
Other than that I see a couple minor issues we could fix before we publish it:

1) A way to turn off search until the docs goes live.
2) Same with News
3) Fix the links URLs in the header
4) "The Hugo Gopher is based on an original work by Renée French." I see no Gopher, so really no need for credit?
5) This may sound pity, but we've had this discussion before. The text "By spf13 and friends" should be changed to something more generic, ala "By the Hugo Authors". Steve is the founder, but that sentence does not capture the truth anymore.

bep commented:
As to the tag fix:

```
-              <span class="f4 mt2 mt0-l db dib-l">({{ with $.Data.Singular }}{{ . | humanize }}:{{ end }} {{ .context.Title }})</span>
+              <span class="f4 mt2 mt0-l db dib-l">({{ with $.context.Data.Singular }}{{ . | humanize -}}: ```

Commit:
fix tags to include context

Commit:
theme changes relative to comments

bep commented on layouts/partials/pagelayout.html:
```diff
@@ -6,7 +6,7 @@
             Hugo Themes
             {{ if not .context.IsHome -}}
-              <span class="f4 mt2 mt0-l db dib-l">({{ with $.Data.Singular }}{{ . | humanize }}:{{ end }} {{ .context.Title }})</span>
+              <span class="f4 mt2 mt0-l db dib-l">({{ with $.context.Data.Singular }}{{ . | humanize }}:{{ end }} {{ .context.Title }})</span>
```
You should also move the space before the end. I'm pretty sure I had that in my example. Now, if the with is false, there will be an extra space, so "( Blog)".

Commit:
temporary adjustments to header

these can be reverted once the sites are integrated

Commit:
add twitter handle to params

Commit:
fix spacing on tags

budparr commented:
I'm not sure what you meant by this "Fix the links URLs in the header" though I may have done this as part of the other changes.

> "The Hugo Gopher is based on an original work by Renée French." I see no Gopher, so really no need for credit?

It does show up on the 404 page and the home page, so I think it's better to leave in.

Commit:
fix the tag spacing in the theme, too

Commit:
make prev next links more intuitive

bep commented:
Yea, the Docs link pointed wrong, but now it is all good -- but now the images do not show up (i.e.  they exist, but does not show) ... Is this only me?

bep commented:
I see some related JS errors:

```
app.bundle.js:8 Uncaught Error: Error: No input element in the page matches #search-input
    at Function.value (app.bundle.js:8)
    at new e (app.bundle.js:8)
    at t (app.bundle.js:15)
    at Object.<anonymous> (app.bundle.js:1)
    at t (app.bundle.js:1)
    at Object.<anonymous> (app.bundle.js:1)
    at t (app.bundle.js:1)
    at app.bundle.js:1
    at app.bundle.js:1
```

budparr commented:
Looks like the JS errors are because I removed search. I don't think that's the source of the image issue.

budparr commented:
Oh, actually, I take that back. It might be related to the script because I'm lazy-loading the images. I think I'll take the simple way out and just hide the search box instead of not rendering it.

Commit:
take a different approach to hiding search

not rendering caused JS errors, so we'll just hide it with CSS

Commit:
theme tweak on prev/next links

budparr commented:
@bep I'm working on prev/next links. I can hold off and do in a different PR, if you like, though it shouldn't be more than a few minutes.

bep commented:
No need to do it in a different PR. Just shout my name when you feel all is done. No rush.

bep commented:
One more thing: There is a "flickering effect" when you load a new page -- like the images are loaded twice. I don't see this in the current theme site, and it looks kind of funky. Looks like some HTTP 301 from "//images" to "/images".

Commit:
make main page listing consistent with single page prev next

Commit:
Fix URLs

Commit:
Remove animation delay because it's causing flash

budparr commented:
@bep I fixed the URLs but I think the flickering effect was caused by an animation delay* I had in there, which I removed, if you could let me know if it's improved.

_* The animation is there because lazyloading shows the image title then the image, and I'd like it to be a smoother transition._

bep commented:
It looks perfect now, thanks.

bep commented:
But I think your sort change from Lastmod to Date is wrong.  We want the most recently updated themes first (we pick the dates from the Git commit log)

budparr commented:
> But I think your sort change from Lastmod to Date is wrong. We want the most recently updated themes first (we pick the dates from the Git commit log)

I was going to ping you in gitter about that. Wrong of me to unilaterally make that change. My thinking was that the previous/next links should be consistent with how the main page is loaded, and to my knowledge there's no way to make them order like that. It's slightly discombulating to go from the main page then flip through the single pages that are in a different order. Only slightly though, so whatever is best; I'm happy to revert that change.

bep commented:
> and to my knowledge there's no way to make them order like that.

There is, but it is badly undocumented. Steve added a feature some years ago, but the API is fairly confusing. But whatever you do, sort by Lastmod.

bep commented:
Also: Having all the images in full width on the theme detail page is maybe not perfect. Look at http://localhost:1313/docuapi/

Commit:
revert change to order by date instead of lasmod

Commit:
theme updates

Commit:
fix issue where images were rendering at full width

budparr commented:
@bep I fixed the images. It was something I didn't understand about the css I was using.

I reverted the main page sorting. I took a glance around the Hugo repo and don't see anything about using prev/next in that context. Happy to take care of it if you can point me in the right direction.

Thanks for your assist on all these. Much appreciated.

bep commented:
@budparr just leave the next/prev as is, it is on my list of things to fix in Hugo, so let us not show some undocumented API to the world and then remove it in a month ...

bep commented:
I will not be able to do this tonight, but if you are happy with the current version (I will also take a quick lookover), I can do the merge and the adjustments to the scripts (we now have a submodule) and push the new version live tomorrow? 

In an ideal world, we should maybe get all sites out at once, but I think it could be motivational to get things moving a little. There have been too much talk and little walk on Gitter.

budparr commented:
Sounds good to me, thank you.
